migration_utils/src/main/kotlin/com/pubnub/migration/App.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * This source file was generated by the Gradle 'init' task
+ */
+package com.pubnub.migration
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.core.Context
+import com.github.ajalt.clikt.core.main
+import com.github.ajalt.clikt.parameters.arguments.argument
+import com.github.ajalt.clikt.parameters.options.flag
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.types.path
+
+class Migration(private val replacements: List<Pair<String, String>>) : CliktCommand("migration_utils") {
+    val dryRun by option(help = "Perform a dry run (without changing any files)").flag()
+    val noBackups by option(help = "Do not save backups of changed files as '<filename>.bck'").flag()
+    val directory by argument(name = "project directory").path(mustExist = true, canBeDir = true, canBeFile = false)
+
+    val supportedExtensions = setOf("java", "kt")
+
+    override fun help(context: Context): String {
+        return "This command will scan all files ending with ${supportedExtensions.map { "'.$it'" }.joinToString(", ")}" +
+                " in <project directory>, and replace package names of classes that have been moved to " +
+                "com.pubnub.api.java.* in PubNub Java SDK v10.\n\n" +
+                "Perform a --dry-run to print a list of files that will be changed.\n\n" +
+                "It is the responsibility of the user to back up any files before running this tool."
+    }
+
+    override val printHelpOnEmptyArgs: Boolean = true
+
+    override fun run() {
+        directory.toFile().walk().forEach { file ->
+            if (file.extension.lowercase() !in supportedExtensions) {
+                return@forEach
+            }
+            val newFileContent = file.useLines { lines ->
+                replaceLines(lines, replacements)
+            }
+            if (newFileContent != null) {
+                if (!dryRun) {
+                    if (!noBackups) {
+                        file.copyTo(file.resolveSibling(file.name + ".bck"), overwrite = false)
```

Comment:
In case .bck file exists exception will be thrown? Is this ok?

Reply:
yes, that's the expected behavior. I don't want to overwrite anything for the user by mistake